### PR TITLE
Add auto-incrementing tokenId, totalSupply functions.  Cleaned up tests and code

### PR DESCRIPTION
**Pull request by plaird523**

### CODE REVIEW

**Review by neodaoist — CHANGES_REQUESTED** [review]

File: src/1729Essay.sol [hunk 1]
```diff
@@ -22,22 +20,23 @@ import {Ownable} from "openzeppelin-contracts/access/Ownable.sol";
 /// @title A collection of winning essays from 1729Writers
 /// @author neodaoist, plaird
 /// @notice A 1729Writers admin can mint and burn essay NFTs on this contract
-/// @dev XYZ
 contract OneSevenTwoNineEssay is Ownable, ERC721, ERC2981 {
```
[the remark this answers is not in the document]
Comment (plaird523) [reply src/1729Essay.sol:23]: Discussed in standup.  Using the ERC2981 implementation from OpenZepplin, confirmed by unit tests

File: src/1729Essay.sol [hunk 2]
```diff
@@ -1,13 +1,11 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.15;
 
-//import {IERC2981} from "openzeppelin-contracts/contracts/interfaces/IERC2981.sol";
 import {ERC721} from "openzeppelin-contracts/token/ERC721/ERC721.sol";
 import {ERC721URIStorage} from "openzeppelin-contracts/token/ERC721/extensions/ERC721URIStorage.sol";
```
[the remark this answers is not in the document]
Comment (plaird523) [reply src/1729Essay.sol:5]: Good catch

File: src/1729Essay.sol [hunk 3]
```diff
@@ -57,69 +53,40 @@ contract OneSevenTwoNineEssay is Ownable, ERC721, ERC2981 {
...
-
+    /// @notice Mint a new token, using the next available token ID
+    /// @param author the address of the writer, who will receive royalty payments
+    /// @param url the URL containing the essay JSON metadata
+    /// @return the tokenId for the newly minted token
+    function mint(address author, string calldata url) public onlyOwner returns (uint256) {
+        uint256 tokenId = nextTokenId.current();
+        nextTokenId.increment();
+        _mint(tokenId, author, url);
+        return tokenId;
```
[the remark this answers is not in the document]
Comment (plaird523) [reply src/1729Essay.sol:64]: Seems unnecessary if the return statement is the last line of the function body, but if it makes you happier I'll add it :grinning:

File: test/1729Essay.t.sol [hunk 4]
```diff
@@ -549,101 +530,136 @@ contract OneSevenTwoNineEssayTest is Test {
...
 
         vm.expectRevert("ERC721: transfer to non ERC721Receiver implementer");
 
-        token.safeTransferFrom(multisig, to, 1337, "testing 456");
+        token.safeTransferFrom(multisig, to, 1, "testing 456");
     }
+
+    // Contract should NOT support receiving funds directly
+    function testReceiveFundsDisabled() public {
+        vm.startPrank(multisig);
```
[the remark this answers is not in the document]
Comment (plaird523) [reply test/1729Essay.t.sol:631]: I like to put the expectation of failure as close to the line that's expected to cause the failure as the framework allows (the line right before in this case).  I think that makes it more clear what's going on when you maintain the test.

All other changes made.

**Review by neodaoist — APPROVED** [review]
LGTM

**Pull request merged**